ongoing monitoring turning on and off

tomer-shvadron opened the pull request:
- **Bal 3305 - improve UI messages for no indications detected (#2936)**
- **fix(report-tabs): hide violations for ads and social media tab (#2937)**
- **feat(merchant-monitoring): display total items in results badge (#2938)**
- **refactor(backoffice-v2): changed home statistics (#2935)**
- **feat: monthly filter on statistics page and additional merchant related data (BAL-3302, BAL-3303) (#2940)**
- **feat(monitoring): implement business monitoring feature**


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

## Release Notes

- **New Features**
  - Added business monitoring functionality with the ability to turn monitoring on and off for merchants
  - Introduced portfolio analytics and risk statistics in the statistics page
  - Enhanced merchant monitoring table with new columns and monitoring status display

- **Improvements**
  - Updated localization for toast messages related to business monitoring
  - Improved date selection and metrics filtering in statistics
  - Enhanced user registration and activity tracking

- **Bug Fixes**
  - Refined portfolio risk statistics display
  - Improved error handling for monitoring state changes

- **Documentation**
  - Updated API schemas to support new monitoring and metrics features

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

tomer-shvadron committed:
feat(monitoring): implement business monitoring feature

- Add success and error messages for turning monitoring on/off
- Update API endpoints to manage ongoing monitoring status
- Integrate tooltip UI for monitoring status display

(Your code is so reactive that I'm surprised it doesn't require a safe word)

tomer-shvadron committed:
chore(business-report): remove unused import for BusinessReportMetricsDto

- Eliminate unnecessary import to clean up the code
- Reduces clutter and potential confusion in the module

(your code is so tidy now, it could be a minimalist's dream home)

Omri-Levy commented on apps/backoffice-v2/src/pages/MerchantMonitoringBusinessReport/mutations/useTurnMonitoringOffMutation/useTurnMonitoringOffMutation.tsx:
```diff
@@ -0,0 +1,44 @@
+  onSuccess,
+}: {
+  onSuccess?: <TData>(data: TData) => void;
+}) => {
```
```suggestion
} = {}) => {
```

Omri-Levy commented on apps/backoffice-v2/src/pages/MerchantMonitoringBusinessReport/mutations/useTurnMonitoringOnMutation/useTurnMonitoringOnMutation.tsx:
```diff
@@ -0,0 +1,44 @@
+  onSuccess,
+}: {
+  onSuccess?: <TData>(data: TData) => void;
+}) => {
```
```suggestion
} = {}) => {
```

Omri-Levy commented on services/workflows-service/src/business-report/business-report.controller.external.ts:
```diff
@@ -111,6 +114,38 @@ export class BusinessReportControllerExternal {
+
+    const reports = await Promise.all(
+      data.map(async report => {
+        const business = await this.businessService.getById(
+          report.merchantId,
+          { select: { metadata: true } },
+          [currentProjectId],
+        );
```
Why are we not using find many over find by id?

Omri-Levy commented on services/workflows-service/src/business-report/business-report.controller.external.ts:
```diff
@@ -224,9 +259,32 @@ export class BusinessReportControllerExternal {
+    return {
+      ...report,
+      monitoringStatus:
+        (isOngoingEnabledForBusiness ||
+          (isOngoingEnabledForBusiness === undefined &&
+            features?.ONGOING_MERCHANT_REPORT?.options?.runByDefault)) ??
+        false,
```
I think this would be easier to read if the conditions were separated or if if conditions were in use.

tomer-shvadron committed:
fix(report): optimize business report fetching logic

- Simplify monitoring mutation definitions
- Consolidate business fetching to reduce database calls

(Your database queries are so chatty, they could use a good night's sleep)
